packages/components/src/site-thumbnail/index.tsx

```diff
@@ -0,0 +1,56 @@
+import classnames from 'classnames';
+import { ReactNode } from 'react';
+import './style.scss';
+import * as Mshots from './use-mshots';
+import { getTextColorFromBackground } from './utils';
+declare let require: ( path: string ) => typeof Mshots;
+
+const MSHOTS_OPTION = {
+	vpw: 1200,
+	vph: 1200,
+	w: 300,
+};
+
+type Props = {
+	backgroundColor?: string;
+	mShotsUrl?: string;
+	size?: string;
+	children?: ReactNode;
+};
+
+export const SiteThumbnail = ( {
+	backgroundColor,
+	children,
+	mShotsUrl,
+	size = 'small',
+}: Props ) => {
+	let useMshots: typeof Mshots;
+	let maybeImage: HTMLImageElement | undefined;
+
+	if ( mShotsUrl ) {
+		useMshots = require( './use-mshots' );
+		maybeImage = useMshots.useMshotsImg( mShotsUrl, MSHOTS_OPTION );
+	}
+
+	const src: string = maybeImage?.src || '';
+	const visible = !! src;
+	const textColor = backgroundColor && getTextColorFromBackground( backgroundColor );
+	const className = classnames(
+		'site-thumbnail',
+		visible ? 'site-thumbnail-visible' : '',
+		`site-thumbnail__size-${ size }`
+	);
+
+	const backgroundImage = maybeImage?.src && `url( ${ maybeImage?.src } )`;
+
+	return ! visible ? (
+		<div className={ className } style={ { backgroundColor: backgroundColor, color: textColor } }>
+			{ children }
+		</div>
+	) : (
+		<div
+			className={ className }
```

Comment:
The generated image renders better as a backgroundImage than using an img tag, img seems to blur the image alot. backgroundImage won't be accessible to screen readers so we have to decide what's the best here.

Reply:
Technically it's not a background but an image. It is against web standards and blocks some use cases, e.g., we can't open such an image in a new window, we can't download it. Screen readers won't catch it as well.

Would it be possible to use an `img` tag and fix rendering issues?

Reply:
Yeah I recognise it's against certain standard depending on the usecase, in our case an `img` might be best to conform to the standards you mention especially from an accessibility standpoint. I'm not sure why the image quality is lost using `img` but will look into it.